tiledb/sm/consolidator/fragment_consolidator.cc

```diff
@@ -163,6 +164,103 @@ Status FragmentConsolidator::consolidate(
   return Status::Ok();
 }
 
+Status FragmentConsolidator::consolidate_fragments(
+    const char* array_name,
+    EncryptionType encryption_type,
+    const void* encryption_key,
+    uint32_t key_length,
+    const std::vector<std::string>& fragment_uris) {
+  auto timer_se = stats_->start_timer("consolidate_frags");
+
+  // Open array for reading
+  Array array_for_reads(URI(array_name), storage_manager_);
+  RETURN_NOT_OK(array_for_reads.open_without_fragments(
+      encryption_type, encryption_key, key_length));
+
+  // Open array for writing
+  Array array_for_writes(array_for_reads.array_uri(), storage_manager_);
+  RETURN_NOT_OK(array_for_writes.open(
+      QueryType::WRITE, encryption_type, encryption_key, key_length));
+
+  // Check if there is anything to consolidate
+  if (fragment_uris.size() <= 1)
+    return Status::Ok();
+
+  // Get all fragment info
+  FragmentInfo fragment_info(URI(array_name), storage_manager_);
+  auto st = fragment_info.load(
+      0,
+      utils::time::timestamp_now_ms(),
+      encryption_type,
+      encryption_key,
+      key_length,
+      false);
+  if (!st.ok()) {
+    array_for_reads.close();
+    array_for_writes.close();
+    return st;
+  }
+
+  // Make a set of the uris to consolidate
+  NDRange union_non_empty_domains;
+  std::unordered_set<std::string> to_consolidate_set;
+  for (auto& uri : fragment_uris) {
+    to_consolidate_set.emplace(uri);
+  }
+
+  // Make sure all fragments to consolidate are present
+  // Compute union of non empty domains as we go
+  uint64_t count = 0;
+  auto& domain{array_for_reads.array_schema_latest().domain()};
```

Comment:
Is there any advantage to doing braced initialization here vs assignment? This construct (brace initialize a ref from a ref?) looks odd to me, although that may just be a difference in style preference.

Reply:
Not sure wither, this came from https://github.com/TileDB-Inc/TileDB/pull/3020.

Reply:
> advantage to doing braced initialization here vs assignment

Syntactic clarity. Braced initialization is obviously initialization; it can't be confused with anything else. Initialization with `=` looks like it might be using `operator=`, which it doesn't, but raises the question.

Reply:
My specific quibble is that this is "initializing" a reference. Anyway, very minor, so I can get used to it. But now wondering if we should clarify the preferred construct in the style guide, because of a similar change in the opposite direction in OL7: https://github.com/TileDB-Inc/TileDB/commit/9603c472943a002b8ee72d7dd76f6b7eac4f0a59#diff-9718a96b5a11bcf93e9a2c15de6a1b689792bd728eb35fbacd23c381b5305f31R587

from 
`auto& domain{array_schema_.domain()};`
to
`const Domain& domain = array_schema_.domain();`

Reply:
IIRC that was previous `Domain *` rather than `Domain &`. I didn't change the initialization style at the time.